### DISCUSSION TITLE
Touch Grid Sensor Installation

### DISCUSSION THREAD

**Discussion by abhihjoshi** [google-deepmind/mujoco#1967] [Asking for Help]
Hello, I am currently looking into the grid touch sensor plugin. However, I am a little confused on how to install this plugin. I was wondering if someone could walk me through the steps or point me to the necessary documentation to install this plugin? Also, would it be possible for someone to include how the to get readings from the sensor using the Python Bindings?

**Comment by yuvaltassa** [comment]
Does the sensor not work out of the box?
The official release is supposed to have all the first-party plugins installed...

**Comment by abhihjoshi** [comment]
I think it automatically works in Ubuntu, but not for Mac. Not sure if this was intended behavior?

**Comment by yuvaltassa** [comment]
Are the comments in #313 relevant for your problem?

**Comment by nimrod-gileadi** [comment]
Can you explain under what circumstance you were encountering this?

i.e. was it:

- A C++ or Python program?
- Were you building MuJoCo libraries from source or downloading the DMG?
- Were you building your own Python bindings? If so, what did you pass for MUJOCO_PLUGIN_PATH?

**Comment by abhihjoshi** [comment]
> A C++ or Python program?

This is in a Python program

> Were you building MuJoCo libraries from source or downloading the DMG?

I was building from source

> Were you building your own Python bindings? If so, what did you pass for MUJOCO_PLUGIN_PATH?

Yes, I built the Python bindings myself. I passed in the same as the MUJOCO_PATH. I imagine that we would need to pass in a different path maybe? Would this just be the path to the plugin directory under the MuJoCo root directory?

**Comment by andy-Chien** [comment]
I got similar issue in Ubuntu 22.04, in my case:

A C++ or Python program?
I tried to load the sample touch grid model with simulate app, which model is "model/plugin/sensor/touch_grid.xml."

Were you building MuJoCo libraries from source or downloading the DMG?
Built from source.

Were you building your own Python bindings? If so, what did you pass for MUJOCO_PLUGIN_PATH?
I run the built simulate app directly so I think this param doesn't work in my case.

I built mujoco and export env params using the follow steps:
```
mkdir build && cd build
cmake ..
cmake --build .
cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local/mujoco
sudo cmake --install .
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/mujoco/lib
export PATH=$PATH:/usr/local/mujoco/bin  
```
Then, when I run:
```
simulate ~/.mujoco/mujoco/model/plugin/sensor/touch_grid.xml
```
I got "XML Error: unknown plugin 'mujoco.sensor.touch_grid"

After checking the source code of main progrom of simulate app, I found that it looking for plugin in "mujoco_plugin" folder inside the same directory of built binary file of simulate app.
But there are no built .so file nor "mujoco_plugin folder" in /usr/local/mujoco/bin and ~/.mujoco/build/bin.
The built .so files are in ~/.mujoco/build/lib, bit it doesn't installed to /usr/local/mujoco/lib.
So I just creat the folder it want and copy the .so files into the folder, and it works.
Steps as follow
```
sudo mkdir /usr/local/mujoco/bin/mujoco_plugin
sodo cp ~/.mujoco/build/lib/*.so /usr/local/mujoco/bin/mujoco_plugin
```
I'm not sure if my solution correct or not, but it works for me.
